### ISSUE TITLE
Add feature to strip EXIF data from photos

### ISSUE THREAD

**Issue by ben-utzer** [friendica/friendica#7653]
This is a feature request. 

Please add an option to strip all or some selected EXIF data from photos uploaded, this option should be a user setting with a default setting to strip all EXIF data when the feature is introduced (retroactive default setting to strip all data). 

This is clearly a data safety issue and it needs to be implemented, it is not good to keep GPS locations in photos when publicized online without giving a choice to remove such data. 

Special perk would be to remove all EXIF data from previously uploaded files. 

This might also be applicable for IPTC meta data, but I am not sure about this, as I never had a device that used this data.

EXIF: https://en.wikipedia.org/wiki/Exif
IPTC: https://en.wikipedia.org/wiki/IPTC_Information_Interchange_Model

How to show EXIF data on Linux: https://mikebeach.org/2011/12/17/how-to-retrieve-exif-data-from-the-command-line-in-ubuntu-linux/

**Comment by rebeka-catalina** [comment]
I like the idea! Imagemagick provides the 'strip' function to remove
exif data from image files. Of course in case of jpeg files this means
another recompression.

**Comment by ben-utzer** [comment]
> I like the idea! Imagemagick provides the 'strip' function to remove exif data from image files. Of course in case of jpeg files this means another recompression.

I think the JPEG are recompressed anyway, so we can probably do this in the same run. 

I hope we can get this feature soon, it really bugs me to have that meta data out there. I was somehow in the believe we have such a function, also @tobiasd thought so. But it seems we don't have it.

**Comment by AlfredSK** [comment]
Good idea. But I think we already have an option to remove the GPS (location) data from uploaded images. Wait a minute...

It's in settings - additional features - photo location.

**Comment by ben-utzer** [comment]
> 
> It's in settings - additional features - photo location.
> 

You mean this?
"Aufnahmeort
Die Foto-Metadaten werden ausgelesen. Dadurch kann der Aufnahmeort (wenn
vorhanden) in einer Karte angezeigt werden."

That doesm't remove it and it doesn't work I think.

**Comment by AlfredSK** [comment]
Hmm. The English text is somewhat different...

Photo metadata is normally stripped. This extracts the location (if present) prior to stripping metadata and links it to a map.

**Comment by AlfredSK** [comment]
When the option is disabled the metadata should be stripped from the images. If that doesn't work it is a bug, I think.

**Comment by ben-utzer** [comment]
It doesn't work.

So the feature request is to improve it and the bug would be to fix it.
Improve the intended (according to the english text description) behavior by adding more options or fixing the bug, for me both sounds fine, but this is broken for a long time now.

**Comment by ben-utzer** [comment]
Oh, can someone confirm it is broken?

**Comment by nupplaphil** [comment]
This metadata feature depends on:
- Is the feature per se disabled (by admin) or enabled?
- Is the feature enabled by the user?
- Is the feature enabled by the **system**?
- What's the **default** setting for the feature?

See
https://github.com/friendica/friendica/blob/3e2f0e9ffa60c47f0faecfd472c72439c00dda09/src/Content/Feature.php#L25-L37
used here:
https://github.com/friendica/friendica/blob/f989c331a3fcf7fa34f7111f4ffa2939e04b4600/mod/photos.php#L778

So it's possible that even when you disable the metadata save of pictures, the overall system-config could override it (yet). So maybe it's not broken but overriden by the admin .. I've to check ..

**Comment by annando** [comment]
That code simply stores the latitude and longitude of the EXIF data in the "photo" table. This feature is not meant to remove that data.

The EXIF data removing will be a new feature.

**Comment by MrPetovan** [comment]
So it is a new feature, and it is about privacy, but it will have to wait the next release.

**Comment by ben-utzer** [comment]
> So it is a new feature, and it is about privacy, but it will have to wait the next release.

So it has to wait another release, already had to wait for 12.2019. 😜

**Comment by MrPetovan** [comment]
Sorry about that, apparently we don't know enough about our own software to be able to categorize this issue within 3 months.

**Comment by tobiasd** [comment]
I still think it is a bug, or a badly advertised feature to not strip the EXIF information like @AlfredSK [wrote above](https://github.com/friendica/friendica/issues/7653#issuecomment-534958637).

**Comment by annando** [comment]
It would be a bug if there had been some non working functionality to remove that data.

**Comment by annando** [comment]
Then it is a bug in the description. Nonetheless we should offer this feature.

**Comment by utzer** [comment]
### Bug Description

The geo location removal tick does not do anything it suggests, neither active not inactive. It does not remove the geo location from a photo, this is dangerous as it gives a false security hope. It should be removed or the description should be updated. 

Never the less a function to strip the meta data is needed urgently!  

### Steps to Reproduce


1.  Upload a photo, once with the tick in /settings/features set and once without. 
2.  Check, that photo still has the gps location in the meta data. 
3.  Get frustrated and use another application to remove the gps location data. 

Actual Result:

nothing

Expected Result:

geo information should be stripped. 

### Screenshots

![Screenshot_2023-03-29_20-13-54](https://user-images.githubusercontent.com/8361662/228618580-089d4815-40dd-4703-87ee-5e100c5e60e1.png)


### Platform Info

Friendica Version: 2023.03-rc, current release candidate



Friendica Source: git

PHP version: 8.2.4

SQL version: not applicable.
